tiledb/api/src/query/subarray.rs

```diff
@@ -359,6 +359,38 @@ where
         Ok(b)
     }
 
+    pub fn set_subarray<IntoRange: Into<Range>>(
+        self,
+        range: IntoRange,
+    ) -> TileDBResult<Self> {
+        let c_subarray = *self.raw;
+
+        let range = range.into();
+        // TODO: should we be checking dimensionality here?
```

Comment:
Have some questions here...

Reply:
I'm basing the following interpretation off of the example `multi_attribute.c`:
```
  // Slice only rows 1, 2 and cols 2, 3, 4
  tiledb_subarray_t* subarray;
  tiledb_subarray_alloc(ctx, array, &subarray);
  int subarray_v[] = {1, 2, 2, 4};
  tiledb_subarray_set_subarray(ctx, subarray, subarray_v);
```

I think what the C function `tiledb_subarray_set_subarray` does is
(a) assume that all dimensions are the same type (which for a dense array they are)
(b) assume that the user input contains two coordinates per dimension
(c) for dimension `d`, set the lower bound of the subarray for that dimension to be the input array index `2 * d` and the upper bound to be array index `2 * d + 1`


So, what you have here works for one dimension since the input is one range, but I imagine for arrays with more dimensions there will be undefined behavior.  Hence to directly answer the question - yes, this definitely should check that the number of input ranges matches the dimensionality!

If you change the input to be a `Vec<Range>` or `Iterator<Item = Range>` or whatever, then you can do a bunch of work to buffer the coordinates and then call `ffi::tiledb_subarray_set_subarray`... but you could also do something like `.enumerate().fold(self, |subarray, (dimension, range)| self.add_range(dimension, range))` and avoid type checks and buffering and etc.

Reply:
Is it correct to say you are suggesting to shim this into a call to .add_range()?

Reply:
Yep.  I think we should add `tiledb_subarray_set_subarray` to the ignored function list.